boj/gold/빵집/jaeung.js

```diff
@@ -0,0 +1,55 @@
+const fs = require("fs");
+const input = fs.readFileSync("dev/stdin").toString().trim().split("\n");
+const [R, C] = input.shift().split(" ").map(Number);
+const grids = input.map((str) => str.split(""));
+
+console.log(solution(R, C, grids));
+
+function solution(R, C, grids) {
+  const visited = Array.from(Array(R), () => Array(C).fill(false));
+  let pipeCount = 0;
+
+  for (let x = 0; x < R; x++) {
+    if (isPlumbing(x, visited, grids)) pipeCount++;
+  }
+
+  return pipeCount;
+}
+
+function isPlumbing(x, visited, grids) {
+  const stack = [[x, 0]];
+  const dx = [1, 0, -1];
+  const dy = [1, 1, 1];
+
+  visited[x][0] = true;
+
+  while (stack.length > 0) {
+    const [currentX, currentY] = stack.pop();
+    const isEnd = currentY === C - 1;
+
+    if (isEnd) {
+      stack.forEach(([x, y]) => (visited[x][y] = false));
+      return true;
+    }
+
+    for (let i = 0; i < dx.length; i++) {
+      const nx = currentX + dx[i];
+      const ny = currentY + dy[i];
+      const isVisit = isBound(nx, ny, R, C) && visited[nx][ny] === false;
+      const isBuilding = isBound(nx, ny, R, C) && grids[nx][ny] === "x";
+
+      if (isVisit && !isBuilding) {
+        stack.push([nx, ny]);
+        visited[nx][ny] = true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function isBound(x, y, R, C) {
+  if (x >= 0 && y >= 0 && x < R && y < C) return true;
+
+  return false;
+}
```

Comment:
ask;
Two Dots 문제에서는 해당 함수에서 삼항 연산자를 이용하여 값을 반환한 것으로 알고 있습니다. `(x >= 0 && y >= 0 && x < R && y < C)` 코드가 값으로 평가되는 문으로 알고 있기 때문에 저는 아래와 같이 작성하는게 더 직관적이라고 생각합니다. 재웅님께서는 조건의 결과를 반환하는 함수는 어떻게 작성하는게 좋다고 생각하시나요?
```suggestion
function isBound(x, y, R, C) {
  return (x >= 0 && y >= 0 && x < R && y < C);
}
```

Reply:
저는 코드를 작성할 때 최대한 흐름을 읽기 편한 쪽으로, 즉 **사람 관점에서 코드를 작성하려고 노력하는 편**입니다.

실제로 제가 생각한 내용을 글쓰기 하듯이 작성하는 편이라, ***어떠한 조건인 경우 이렇게, 아닌 경우 저렇게 한다*** 처럼 말로 하면서 작성을 하다보니, 제시해주신 표현식이나 `DFS`에서 자주 사용하는 방문 배열의 확인 유무도  `const isVisit = visited[index] === true`처럼 작성하게 되더라구요..😅

질문하신 내용에 대한 답변으로는 ***`isBound(....)`라는 이름의 함수 자체가 네이밍을 통해 `boolean` 값을 반환한다는 것을 알 수 있으니, 함수의 로직이 복잡하지 않다면 둘이 큰 차이가 없지 않을까?*** 라고 생각됩니다!